cmd/frankenstein/main.go

```diff
@@ -53,126 +54,138 @@ func init() {
+	flag.StringVar(&cfg.consulHost, "consul.hostname", "localhost:8500", "Hostname and port of Consul.")
+	flag.StringVar(&cfg.consulPrefix, "consul.prefix", "collectors/", "Prefix for keys in Consul.")
+	flag.StringVar(&cfg.s3URL, "s3.url", "localhost:4569", "S3 endpoint URL.")
+	flag.StringVar(&cfg.dynamodbURL, "dynamodb.url", "localhost:8000", "DynamoDB endpoint URL.")
+	flag.BoolVar(&cfg.dynamodbCreateTables, "dynamodb.create-tables", false, "Create required DynamoDB tables on startup.")
+	flag.StringVar(&cfg.memcachedHostname, "memcached.hostname", "", "Hostname for memcached service to use when caching chunks. If empty, no memcached will be used.")
+	flag.DurationVar(&cfg.memcachedTimeout, "memcached.timeout", 100*time.Millisecond, "Maximum time to wait before giving up on memcached requests.")
+	flag.DurationVar(&cfg.memcachedExpiration, "memcached.expiration", 0, "How long chunks stay in the memcache.")
+	flag.StringVar(&cfg.memcachedService, "memcached.service", "memcached", "SRV service used to discover memcache servers.")
+	flag.DurationVar(&cfg.remoteTimeout, "remote.timeout", 5*time.Second, "Timeout for downstream ingesters.")
+	flag.DurationVar(&cfg.flushPeriod, "ingester.flush-period", 1*time.Minute, "Period with which to attempt to flush chunks.")
+	flag.DurationVar(&cfg.maxChunkAge, "ingester.max-chunk-age", 10*time.Minute, "Maximum chunk age before flushing.")
+	flag.IntVar(&cfg.numTokens, "ingester.num-tokens", 128, "Number of tokens for each ingester.")
 	flag.Parse()
 
-	consul, err := frankenstein.NewConsulClient(consulHost)
+	chunkStore, err := setupChunkStore(cfg)
 	if err != nil {
-		log.Fatalf("Error initializing Consul client: %v", err)
+		log.Fatalf("Error initializing chunk store: %v", err)
 	}
-	consul = frankenstein.PrefixClient(consul, consulPrefix)
 
-	var chunkCache *frankenstein.ChunkCache
-	if memcachedHostname != "" {
-		chunkCache = &frankenstein.ChunkCache{
-			Memcache: frankenstein.NewMemcacheClient(frankenstein.MemcacheConfig{
-				Host:           memcachedHostname,
-				Service:        memcachedService,
-				Timeout:        memcachedTimeout,
-				UpdateInterval: 1 * time.Minute,
-			}),
-			Expiration: memcachedExpiration,
-		}
-	}
-	chunkStore, err := frankenstein.NewAWSChunkStore(frankenstein.ChunkStoreConfig{
-		S3URL:       s3URL,
-		DynamoDBURL: dynamodbURL,
-		ChunkCache:  chunkCache,
-	})
+	consul, err := ring.NewConsulClient(cfg.consulHost)
 	if err != nil {
-		log.Fatal(err)
-	}
-	if dynamodbCreateTables {
-		if err = chunkStore.CreateTables(); err != nil {
-			log.Fatal(err)
-		}
+		log.Fatalf("Error initializing Consul client: %v", err)
 	}
+	consul = ring.PrefixClient(consul, cfg.consulPrefix)
 
-	switch mode {
+	switch cfg.mode {
 	case distributor:
-		setupDistributor(consul, chunkStore, remoteTimeout)
+		ring := ring.NewRing(consul)
+		defer ring.Stop()
+		setupDistributor(cfg, ring, chunkStore)
+		if err != nil {
```

Comment:
`setupDistributor` is returning two values `(*Ring, err)` that are being thrown away here. I think you meant to at least check that the returned error is not nil, rather than whatever happens to be in `err` before this is called.